packages/openapi-generator/src/document-converter.ts

```diff
@@ -73,13 +79,168 @@ export function convertDocToGlobalTag(
+} {
+  return namedOperations.reduce(
+    (
+      partitionedOperations: {
+        uniqueOperationNames: string[];
+        operationsWithDuplicateNames: OpenAPIV3.OperationObject[];
+      },
+      operation
+    ) => {
+      if (
+        partitionedOperations.uniqueOperationNames.includes(
+          operation.operationId!
+        )
+      ) {
+        partitionedOperations.operationsWithDuplicateNames.push(operation);
+      } else {
+        partitionedOperations.uniqueOperationNames.push(operation.operationId!);
+      }
+      return partitionedOperations;
+    },
+    { uniqueOperationNames: [], operationsWithDuplicateNames: [] }
+  );
+}
+
+function partitionOperationsToNamedAndUnnamed(
+  openApiDocument: OpenAPIV3.Document
+): {
+  namedOperations: OpenAPIV3.OperationObject[];
+  unnamedOperationsWithAdditionalInfo: OperationWithAdditionalInfo[];
+} {
+  const namedOperations: OpenAPIV3.OperationObject[] = [];
+  const unnamedOperationsWithAdditionalInfo: OperationWithAdditionalInfo[] = [];
+  executeForAllOperationObjects(
+    openApiDocument,
+    (operation, pathPattern, method) => {
+      if (operation.operationId) {
+        namedOperations.push(operation);
+      } else {
+        unnamedOperationsWithAdditionalInfo.push({
+          operation,
+          pathPattern,
+          method
+        });
+      }
+    }
+  );
+
+  return {
+    namedOperations,
+    unnamedOperationsWithAdditionalInfo
+  };
+}
+
+function setUniqueOperationName(
+  operation: OpenAPIV3.OperationObject,
+  name: string,
+  nameGenerator: UniqueNameGenerator
+) {
+  operation.operationId = nameGenerator.generateAndSaveUniqueName(name);
+}
+
+function executeForAllOperationObjects(
+  openApiDocument: OpenAPIV3.Document,
+  callback: (
+    operation: OpenAPIV3.OperationObject,
+    pathPattern: string,
+    method: Method
+  ) => any
+): void {
+  return Object.entries(openApiDocument.paths).forEach(
+    ([pathPattern, pathDefinition]: [string, OpenAPIV3.PathItemObject]) => {
+      methods.forEach(method => {
+        if (pathDefinition[method]) {
+          callback(pathDefinition[method]!, pathPattern, method);
+        }
+      });
+    }
+  );
+}
+
+export function getOperationNameFromPatternAndMethod(
+  pattern: string,
+  method: Method
+): string {
+  const [placeholders, pathParts] = partition(pattern.split('/'), part =>
+    /^\{.*\}$/.test(part)
+  );
+  const prefix = getSpeakingNameForMethod(method).toLowerCase();
+  const base = pathParts.map(part => pascalCase(part)).join('');
+  const suffix = placeholders.length
+    ? 'By' + placeholders.map(part => pascalCase(part)).join('And')
+    : '';
+
+  return prefix + base + suffix;
+}
+
+function getSpeakingNameForMethod(method: Method): string {
+  return nameMapping[method];
+}
+
+const nameMapping = {
+  get: 'get',
+  post: 'create',
+  patch: 'update',
+  put: 'update',
+  delete: 'delete',
+  head: 'getHeadersFor',
+  options: 'getOptionsFor',
+  trace: 'trace'
```

Comment:
I guess you did a manual test with the openapi generator and made this alignment.
Could we add a follow up ticket, so we can have a test that checking this alignment or maybe it's covered by some tests implicitly?

Reply:
What exactly do you mean? This is used for the names of the operationIds if they are not given and there is test for some of those (I didn't go through all of them because that would be trivial). Or do you mean an alignment to something else?